tests/potential/test_pair.py

```diff
@@ -230,14 +230,16 @@ def test_derivative(self):
         s = relentless.potential.Spline(types=('1',), num_knots=3)
         s.from_array(pair=('1','1'), r=r_arr, u=u_arr)
         d_actual = np.array([1.125,0.625,0])
-        d = s.derivative(pair=('1','1'), param='knot-1', r=[1.5,2.5,3.5])
+        param = [i for i in s.knots(('1','1'))][1][1]
```

Comment:
You can convert a generator to a sequence via the constructor:
```suggestion
        param = list(s.knots(('1','1')))[1][1]
```